### PR TITLE
GN 169 Dropdown open and close hooks

### PR DESCRIPTION
**Pull request by johndavey72**
https://nicedigital.atlassian.net/browse/GN-169

### CODE REVIEW

**Review by AndyChildNice — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -408,6 +410,53 @@ var global_nav_config = {
...
+  },
+};
+```
+
+###### Header.onDropdownClose
+
+- Type: `String`, `Function`
+- Default: `null`
+
+Function parameters:
```
Comment (ediblecode) [inline README.md:439]: Can you remove thus function parameters section here please as there are no args for these new hooks

File: src/Header/Header.jsx [hunk 2]
```diff
@@ -77,12 +77,48 @@ export class Header extends Component {
...
+			}
+		}
+
+		if (onDropdownClose) {
+			const onDropdownCloseCallback =
+				typeof onDropdownClose === "function"
+					? onDropdownClose
+					: window[onDropdownClose];
+			if (
+				idOfOpenDropdown == null &&
```
Comment (ediblecode) [inline src/Header/Header.jsx:104]: Makes sense for the id check to be in the outer if I think. Same for the open state above

File: src/Header/Header.jsx [hunk 3]
```diff
@@ -77,12 +77,48 @@ export class Header extends Component {
...
+				typeof onDropdownCloseCallback === "function"
+			) {
+				onDropdownCloseCallback();
+				return true;
+			}
+		}
+
+		return false;
```
Comment (ediblecode) [inline src/Header/Header.jsx:112]: I'm not sure we need these `return` statements in here do we?

File: src/Header/Header.test.jsx [hunk 4]
```diff
@@ -137,6 +137,28 @@ describe("Header", () => {
 		});
 	});
 
+	describe("Dropdown open and close callbacks", () => {
+		it("should call onDropdownOpen callback prop if dropdown is opened", () => {
```
Comment (ediblecode) [inline src/Header/Header.test.jsx:141]: Is it worth adding tests for global functions, e.g. when passing in a string as the name of a function?

File: README.md [hunk 5]
```diff
@@ -42,6 +42,8 @@
 					- [Header.skipLinkId](#headerskiplinkid)
 					- [Header.onNavigating](#headeronnavigating)
 					- [Header.onResize](#headeronresize)
+					- [Header.onDropdownOpen](#headerondropdownopen)
```
Comment (ediblecode) [inline README.md:45]: We're missing TypeScript defs for the new props. Please can you add them?

File: src/Header/Header.jsx [hunk 6]
```diff
@@ -77,12 +77,48 @@ export class Header extends Component {
 		}
 	}
 
+	dropdownToggleHandler(idOfOpenDropdown) {
+		// If we've got an onDropdownOpen or onDropdownClose prop call the onDropdownOpen or onDropdownClose callback based on idOfOpenDropdown.
+		const { onDropdownOpen, onDropdownClose } = this.props;
+
+		if (onDropdownOpen) {
+			const onDropdownOpenCallback =
```
Comment (ediblecode) [inline src/Header/Header.jsx:85]: I wonder if it's worth abstracting this logic out for checking for a function/name of a function as it's used in a few places?

**Review by ediblecode — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (johndavey72) [reply README.md:439]: ahh yes missed those

**Review by ediblecode — APPROVED** [review]

**Pull request merged**